concurrency/buffer.go

```diff
@@ -5,18 +5,21 @@ import (
 	"sync"
 )
 
+// SyncBuffer is a struct that contains a buffer and a mutex lock to allow for synchronous access to the contained buffer.
```

Comment:
:) That describes implementation more than intent. What about...

```suggestion
// SyncBuffer is an io.Writer implementation with atomic writes. It keeps data only in the memory.
```